tests/recovery/e2e_recovery_test.go

```diff
@@ -0,0 +1,136 @@
+	if err := config.InitConfig(); err != nil {
+		log.WithError(err).Fatal("Could not initialize configuration")
+	}
+
+	if *poseidonBinary == "" {
+		log.Fatal("You must specify the -path to the Poseidon binary!")
+	}
+	if *testDockerImage == "" {
+		log.Fatal("You must specify the -dockerImage flag!")
+	}
+
+	nomadNamespace = config.Config.Nomad.Namespace
+	var err error
+	nomadClient, err = nomadApi.NewClient(&nomadApi.Config{
+		Address:   config.Config.Nomad.URL().String(),
+		TLSConfig: &nomadApi.TLSConfig{},
+		Namespace: nomadNamespace,
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Could not create Nomad client")
+		return
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestE2ERecoveryTests(t *testing.T) {
+	testSuite := new(E2ERecoveryTestSuite)
+
+	ctx, cancelPoseidon := context.WithCancel(context.Background())
+	testSuite.poseidonCancel = cancelPoseidon
+
+	startPoseidon(ctx, cancelPoseidon)
+	waitForPoseidon()
+
+	e2e.CreateDefaultEnvironment(PrewarmingPoolSize, *testDockerImage)
+	e2e.WaitForDefaultEnvironment()
+
+	suite.Run(t, testSuite)
+
+	TearDown()
+	testSuite.poseidonCancel()
+	<-time.After(tests.ShortTimeout)
+}
+
+func (s *E2ERecoveryTestSuite) TestInactivityTimer_Valid() {
+	_, err := e2e.ProvideWebSocketURL(s.runnerID, &dto.ExecutionRequest{Command: "true"})
+	s.NoError(err)
+}
+
+func (s *E2ERecoveryTestSuite) TestInactivityTimer_Expired() {
+	<-time.After(InactivityTimeout * time.Second)
+	_, err := e2e.ProvideWebSocketURL(s.runnerID, &dto.ExecutionRequest{Command: "true"})
+	s.Error(err)
+}
+
+// We expect the runner count to be equal to the prewarming pool size plus the one provided runner.
+// If the count does not include the provided runner, the evaluation of the runner status may be wrong.
+func (s *E2ERecoveryTestSuite) TestRunnerCount() {
+	jobListStubs, _, err := nomadClient.Jobs().List(&nomadApi.QueryOptions{
+		Prefix:    tests.DefaultEnvironmentIDAsString,
+		Namespace: nomadNamespace,
+	})
+	s.Require().NoError(err)
+	s.Equal(PrewarmingPoolSize+1, len(jobListStubs))
+}
+
+func (s *E2ERecoveryTestSuite) TestEnvironmentStatistics() {
+	url := helpers.BuildURL(api.BasePath, api.StatisticsPath, api.EnvironmentsPath)
+	response, err := http.Get(url) //nolint:gosec // The variability of this url is limited by our configurations.
+	s.Require().NoError(err)
+	s.Require().Equal(http.StatusOK, response.StatusCode)
+
+	statistics := make(map[string]*dto.StatisticalExecutionEnvironmentData)
+	err = json.NewDecoder(response.Body).Decode(&statistics)
+	s.Require().NoError(err)
+	err = response.Body.Close()
+	s.Require().NoError(err)
+
+	environmentStatistics, ok := statistics[tests.DefaultEnvironmentIDAsString]
+	s.Require().True(ok)
+	s.Equal(tests.DefaultEnvironmentIDAsInteger, environmentStatistics.ID)
+	s.Equal(uint(PrewarmingPoolSize), environmentStatistics.PrewarmingPoolSize)
+	s.Equal(uint(PrewarmingPoolSize), environmentStatistics.IdleRunners)
+	s.Equal(uint(1), environmentStatistics.IdleRunners)
```

Comment:
Shouldn't 1 be equal to the `usedRunners` instead of the `idleRunners` (where we have two)?

This test also fails in [our pipeline](https://github.com/openHPI/poseidon/actions/runs/3549111579/jobs/5969421170).